File: vue/src/api/gameNews.ts
/**
 * 游戏新闻 API
 */
import http from './http';

export interface GameNewsItem {
  gid: string;
  title: string;
  url: string;
  isExternalUrl: boolean;
  author: string;
  contents: string;
  feedLabel: string;
  date: number;
}

export interface GameNewsResult {
  appid: number;
  items: GameNewsItem[];
}

function unwrap<T>(response: any): T {
  return response.data?.data ?? response.data ?? ({} as T);
}

/** GET /api/game-news */
export async function getGameNews(appid: number, count = 10, maxlength?: number): Promise<GameNewsResult> {
  const params: Record<string, any> = { appid, count };
  if (maxlength != null) params.maxlength = maxlength;
  const response = await http.get('/game-news', { params });
  return unwrap<GameNewsResult>(response);
}

File: steam-api/src/main/java/com/SteamGame/api/domain/news/GameNews.java
package com.SteamGame.api.domain.news;

import java.sql.Timestamp;

/**
 * Domain entity for game_news table.
 * Caches game news from ISteamNews/GetNewsForApp.
 */
public class GameNews {
    private String gid;
    private Long appid;
    private String title;
    private String url;
    private Boolean externalUrl;
    private String author;
    private String contents;
    private String feedLabel;
    private Long date;
    private Timestamp syncedAt;

    public String getGid() { return gid; }
    public void setGid(String gid) { this.gid = gid; }

    public Long getAppid() { return appid; }
    public void setAppid(Long appid) { this.appid = appid; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }

    public Boolean getExternalUrl() { return externalUrl; }
    public void setExternalUrl(Boolean externalUrl) { this.externalUrl = externalUrl; }

    public String getAuthor() { return author; }
    public void setAuthor(String author) { this.author = author; }

    public String getContents() { return contents; }
    public void setContents(String contents) { this.contents = contents; }

    public String getFeedLabel() { return feedLabel; }
    public void setFeedLabel(String feedLabel) { this.feedLabel = feedLabel; }

    public Long getDate() { return date; }
    public void setDate(Long date) { this.date = date; }

    public Timestamp getSyncedAt() { return syncedAt; }
    public void setSyncedAt(Timestamp syncedAt) { this.syncedAt = syncedAt; }
}
